src/client/components/ImageMenu.js

```diff
@@ -102,29 +105,53 @@ const ImageMenu = ({
     imageExists &&
     boxes[ethereumAddress].publicProfile[imageTag][0].contentUrl['/']
 
+  const handleToggleMenu = imageTag => {
+    imageTag === activeImageMenu
+      ? toggleImageMenu('')
+      : toggleImageMenu(imageTag)
+  }
+
   return (
-    <ImageMenuStyled
-      top={top}
-      right={right}
-      dragging={dragState.dragging}
-      {...getRootProps({
-        className: 'dropzone',
-        isDragActive,
-        isDragAccept,
-        isDragReject,
-        imageCid,
-      })}
-    >
-      {' '}
-      <div>Update {imageTitle} photo</div>
-      <input {...getInputProps({ disabled: false })} />
-      <div onClick={open}>Upload new image</div>
-      {imageExists && (
-        <div onClick={() => dispatchModal(removeItem(imageCid, imageTag))}>
-          Delete
+    <div tabIndex="-1" onBlur={() => toggleImageMenu('')}>
+      <ImageMenuStyled
+        top={top}
+        right={right}
+        dragging={dragState.dragging}
+        active={imageTag === activeImageMenu}
+        {...getRootProps({
+          className: 'dropzone',
+          isDragActive,
+          isDragAccept,
+          isDragReject,
+          imageCid,
+        })}
+      >
+        <div
+          css="padding: 8px 8px 6px 8px"
+          onClick={() => handleToggleMenu(imageTag)}
```

Comment:
Let's only add `onClick` to `<a>` and `<button>`

Reply:
I think we can get rid of that `handleToggleMenu` definition by changing this line to

    onClick={() => setActive(!active)}

Reply:
Very good point. It started as multi-component thing, so it needed to distinct various active menus - it's no longer the case.